src/services/safe-apps/manifest.ts

```diff
@@ -99,7 +99,7 @@ const fetchSafeAppFromManifest = async (
   const iconUrl = getAppLogoUrl(normalizedAppUrl, appManifest)
 
   return {
-    id: Math.random(),
+    id: Math.round(Math.random() * Math.pow(10, 18)),
```

Comment:
This leads to a number space that can still include low numbers which are given to actual Safe apps.
Lets add a billion or so to the random number: `id: Math.round(1_000_000_000 + Math.random() * Math.pow(10, 18))` to avoid collisions.